Build fails on Mac OS X 10.9.5

```
Edwards-MacBook-Air:src emv$ git clone https://github.com/opencontainers/runc.git
Cloning into 'runc'...
remote: Counting objects: 7237, done.
remote: Total 7237 (delta 0), reused 0 (delta 0), pack-reused 7237
Receiving objects: 100% (7237/7237), 1.59 MiB | 1.67 MiB/s, done.
Resolving deltas: 100% (4531/4531), done.
Checking connectivity... done.
Edwards-MacBook-Air:src emv$ cd runc
Edwards-MacBook-Air:runc emv$ ls
Godeps		Makefile	checkpoint.go	libcontainer	restore.go	signals.go	spec_linux.go	utils.go
LICENSE		README.md	events.go	main.go		run.go		spec.go		tty.go
Edwards-MacBook-Air:runc emv$ make
go get github.com/tools/godep
godep go build -o runc .
checkpoint.go:9:2: cannot find package "github.com/opencontainers/runc/libcontainer" in any of:
	/usr/local/go/src/pkg/github.com/opencontainers/runc/libcontainer (from $GOROOT)
	/Users/emv/src/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer (from $GOPATH)
	/Users/emv/src/github.com/opencontainers/runc/libcontainer
restore.go:11:2: cannot find package "github.com/opencontainers/runc/libcontainer/configs" in any of:
	/usr/local/go/src/pkg/github.com/opencontainers/runc/libcontainer/configs (from $GOROOT)
	/Users/emv/src/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/configs (from $GOPATH)
	/Users/emv/src/github.com/opencontainers/runc/libcontainer/configs
main.go:10:2: cannot find package "github.com/opencontainers/runc/libcontainer/nsenter" in any of:
	/usr/local/go/src/pkg/github.com/opencontainers/runc/libcontainer/nsenter (from $GOROOT)
	/Users/emv/src/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/nsenter (from $GOPATH)
	/Users/emv/src/github.com/opencontainers/runc/libcontainer/nsenter
restore.go:12:2: cannot find package "github.com/opencontainers/runc/libcontainer/utils" in any of:
	/usr/local/go/src/pkg/github.com/opencontainers/runc/libcontainer/utils (from $GOROOT)
	/Users/emv/src/runc/Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/utils (from $GOPATH)
	/Users/emv/src/github.com/opencontainers/runc/libcontainer/utils
godep: go exit status 1
make: *** [all] Error 1
Edwards-MacBook-Air:runc emv$ uname
Darwin
Edwards-MacBook-Air:runc emv$ 
```

Different error on OS X Yosemite:

```
cd runc/
tim@home /go/src/github.com/opencontainers/runc> make
go get github.com/tools/godep
godep go build -o runc .
# github.com/opencontainers/runc/libcontainer/configs
libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
libcontainer/nsenter/nsexec.c:8:10: fatal error: 'linux/limits.h' file not found
godep: go exit status 2
make: *** [all] Error 1
```

+1 to @AndrewVos  who got me closer with his install instructions. Again on Mac OS X 10.9, some progress:

```
Edwards-MacBook-Air:runc emv$ go get github.com/opencontainers/runc
# github.com/opencontainers/runc/libcontainer/configs
../github.com/opencontainers/runc/libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
../github.com/opencontainers/runc/libcontainer/nsenter/nsexec.c:8:10: fatal error: 'linux/limits.h' file not found
Edwards-MacBook-Air:runc emv$ cd $GOPATH/src/github.com/opencontainers/runc
Edwards-MacBook-Air:runc emv$ make
go get github.com/tools/godep
godep go build -o runc .
# github.com/opencontainers/runc/libcontainer/configs
libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
libcontainer/nsenter/nsexec.c:8:10: fatal error: 'linux/limits.h' file not found
godep: go exit status 2
make: *** [all] Error 1
```

at least that suggests where a change might be made.

Same thing happens on OSX 10.10.3

```
# github.com/opencontainers/runc/libcontainer/configs
libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
libcontainer/nsenter/nsexec.c:8:10: fatal error: 'linux/limits.h' file not found
godep: go exit status 2
make: *** [all] Error 1
```

@vielmetti is onto something here.
It should be `#include <limits.h>` because on OS X Yosemite that file is located here:
```
/Library/Developer/CommandLineTools/usr/lib/clang/6.1.0/include/limits.h
```
Once I changed that to limits.h, I got different errors:
```
make
go get github.com/tools/godep
godep go build -o runc .
# github.com/opencontainers/runc/libcontainer/configs
libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
libcontainer/nsenter/nsexec.c:61:10: warning: implicit declaration of function 'clone' is invalid in C99 [-Wimplicit-function-declaration]
libcontainer/nsenter/nsexec.c:61:42: error: use of undeclared identifier 'CLONE_PARENT'
libcontainer/nsenter/nsexec.c:135:7: warning: implicit declaration of function 'setns' is invalid in C99 [-Wimplicit-function-declaration]
libcontainer/nsenter/nsexec.c:154:8: warning: implicit declaration of function 'dup3' is invalid in C99 [-Wimplicit-function-declaration]
godep: go exit status 2
make: *** [all] Error 1
```

unfortunately @ggiamarchi's snippet fails in mac (Mac OS 10.10.2)
```
godep get github.com/opencontainers/runc
# github.com/opencontainers/runc/libcontainer/configs
libcontainer/configs/namespaces.go:5: undefined: Namespace
# github.com/opencontainers/runc/libcontainer/nsenter
libcontainer/nsenter/nsexec.c:8:10: fatal error: 'linux/limits.h' file not found
godep: go exit status 2
godep: exit status 1
```

--- a/checkpoint.go
+++ b/checkpoint.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (

--- a/events.go
+++ b/events.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (
@@ -56,7 +58,7 @@ var eventsCommand = cli.Command{
 			return
 		}
 		go func() {
-			for _ = range time.Tick(context.Duration("interval")) {
+			for range time.Tick(context.Duration("interval")) {
 				s, err := container.Stats()
 				if err != nil {
 					logrus.Error(err)

--- a/main.go
+++ b/main.go
@@ -2,11 +2,9 @@ package main
 
 import (
 	"os"
-	"runtime"
 
 	"github.com/Sirupsen/logrus"
 	"github.com/codegangsta/cli"
-	"github.com/opencontainers/runc/libcontainer"
 )
 
 const (
@@ -26,18 +24,6 @@ After creating a spec for your root filesystem with runc, you can execute a simp
     runc
 `
 )
-
-func init() {
-	if len(os.Args) > 1 && os.Args[1] == "init" {
-		runtime.GOMAXPROCS(1)
-		runtime.LockOSThread()
-		factory, _ := libcontainer.New("")
-		if err := factory.StartInitialization(); err != nil {
-			fatal(err)
-		}
-		panic("--this line should never been executed, congratulations--")
-	}
-}
 
 func main() {
 	app := cli.NewApp()
@@ -77,23 +63,8 @@ func main() {
 		}
 		return nil
 	}
-	// default action is to execute a container
-	app.Action = func(context *cli.Context) {
-		spec, err := loadSpec(context.Args().First())
-		if err != nil {
-			fatal(err)
-		}
-		if os.Geteuid() != 0 {
-			logrus.Fatal("runc should be run as root")
-		}
-		status, err := execContainer(context, spec)
-		if err != nil {
-			logrus.Fatalf("Container start failed: %v", err)
-		}
-		// exit with the container's exit status so any external supervisor is
-		// notified of the exit with the correct exit status.
-		os.Exit(status)
-	}
+
+	app.Action = runAction
 
 	if err := app.Run(os.Args); err != nil {
 		logrus.Fatal(err)

--- a/main_unsupported.go
+++ b/main_unsupported.go
@@ -1,0 +1,22 @@
+// +build !linux
+
+package main
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+)
+
+func getDefaultID() string {
+	return ""
+}
+
+var (
+	checkpointCommand cli.Command
+	eventsCommand     cli.Command
+	restoreCommand    cli.Command
+)
+
+func runAction(*cli.Context) {
+	logrus.Fatal("Current OS is not supported yet")
+}

--- a/restore.go
+++ b/restore.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (

--- a/run.go
+++ b/run.go
@@ -1,13 +1,28 @@
+// +build linux
+
 package main
 
 import (
 	"fmt"
 	"os"
+	"runtime"
 
 	"github.com/Sirupsen/logrus"
 	"github.com/codegangsta/cli"
 	"github.com/opencontainers/runc/libcontainer"
 )
+
+func init() {
+	if len(os.Args) > 1 && os.Args[1] == "init" {
+		runtime.GOMAXPROCS(1)
+		runtime.LockOSThread()
+		factory, _ := libcontainer.New("")
+		if err := factory.StartInitialization(); err != nil {
+			fatal(err)
+		}
+		panic("--this line should never been executed, congratulations--")
+	}
+}
 
 func execContainer(context *cli.Context, spec *Spec) (int, error) {
 	config, err := createLibcontainerConfig(spec)
@@ -46,6 +61,24 @@ func execContainer(context *cli.Context, spec *Spec) (int, error) {
 		return -1, err
 	}
 	return handler.forward(process)
+}
+
+// default action is to execute a container
+func runAction(context *cli.Context) {
+	spec, err := loadSpec(context.Args().First())
+	if err != nil {
+		fatal(err)
+	}
+	if os.Geteuid() != 0 {
+		logrus.Fatal("runc should be run as root")
+	}
+	status, err := execContainer(context, spec)
+	if err != nil {
+		logrus.Fatalf("Container start failed: %v", err)
+	}
+	// exit with the container's exit status so any external supervisor is
+	// notified of the exit with the correct exit status.
+	os.Exit(status)
 }
 
 func destroy(container libcontainer.Container) {

--- a/signals.go
+++ b/signals.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (

--- a/spec.go
+++ b/spec.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"runtime"
 
+	"github.com/Sirupsen/logrus"
 	"github.com/codegangsta/cli"
 )
 
@@ -111,7 +112,7 @@ var specCommand = cli.Command{
 		}
 		data, err := json.MarshalIndent(&spec, "", "\t")
 		if err != nil {
-			fatal(err)
+			logrus.Fatal(err)
 		}
 		fmt.Printf("%s", data)
 	},

--- a/tty.go
+++ b/tty.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (

--- a/utils.go
+++ b/utils.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package main
 
 import (

